js/bybit.js

```diff
@@ -2337,29 +2791,63 @@ module.exports = class bybit extends Exchange {
         //         }
         //     }
         //
-        let trades = this.safeValue (response, 'result', {});
-        if (!Array.isArray (trades)) {
-            trades = this.safeValue2 (trades, 'dataList', 'list', []);
-        }
+        const result = this.safeValue (response, 'result', {});
+        const trades = this.safeValue (result, 'dataList', []);
         return this.parseTrades (trades, market, since, limit);
     }
 
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name bybit#fetchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the bybit api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        let enableUnifiedMargin = undefined;
+        [ enableUnifiedMargin, params ] = this.handleOptionAndParamsValue (params, 'fetchTrades', 'enableUnifiedMargin', false);
+        const isUsdcSettled = market['settle'] === 'USDC';
+        if (market['type'] === 'spot') {
+            return await this.fetchSpotTrades (symbol, since, limit, params);
+        } else if (!isUsdcSettled) {
+            if (enableUnifiedMargin) {
```

Comment:
We should not check `enableUnifiedMargin` option here, as I said before unified is an orthogonal concept to trades, tickers, markets, etc (only applies to order/account-related methods)

So in this case, I think the logic should be:
-  if v3 version 
   - use v3 endpoint to fetchTrades (I think the same endpoint fetches options and usdc settled as well but pls double check it)
   - otherwise
      - use the current logic

Reply:
@carlosmiei because these endpoints are public, do you suggest we replace v2 calls with v3 derivatives method? For example:

```JS
if (type === 'spot') {
    return await this.fetchSpotTickers (symbols, params);
}
return await this.fetchDerivativesTickers (symbols, params);
```

In this case, I think we can remove `fetchV2DerivativesTickers`.

Reply:
How do you think we only check the version in public methods?

Reply:
@sc0Vu yeah that sounds good